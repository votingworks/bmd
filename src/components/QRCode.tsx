import QRCodeReact from '@votingworks/qrcode.react'
import React from 'react'
import styled from 'styled-components'

const ResponsiveSvgWrapper = styled.div`
  & > svg {
    display: block; /* svg is "inline" by default */
    width: 100%; /* reset width */
    height: auto; /* reset height */
  }
`

interface QRCodeProps {
  value: string | Uint8Array
  size?: number
  bgColor?: string
  fgColor?: string
  level?: 'L' | 'M' | 'Q' | 'H'
  renderAs?: 'svg' | 'canvas'
}

const QRCode = ({ level = 'H', renderAs = 'svg', value }: QRCodeProps) => (
  <ResponsiveSvgWrapper>
    <QRCodeReact renderAs={renderAs} value={value} level={level} />
  </ResponsiveSvgWrapper>
)

export default QRCode
