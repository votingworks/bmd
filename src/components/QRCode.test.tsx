import React from 'react'
import { render } from '@testing-library/react'

import QRCode from './QRCode'

describe('renders QR Code with size', () => {
  it('renders basic QR Code', async () => {
    const { container } = render(
      <QRCode size="1in" value="http://voting.works" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('renders QR Code', async () => {
    const { container } = render(<QRCode value="VX.21.5" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.querySelector('path[fill="#000000"]')!)
      .toMatchInlineSnapshot(`
      <path
        d="M0 0h7v1H0zM8 0h3v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM10 1h1v1H10zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM11 2h1v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h4v1H9zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM12 4h1v1H12zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM9 5h1v1H9zM11 5h2v1H11zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h1v1H8zM12 7h1v1H12zM5 8h2v1H5zM9 8h2v1H9zM14 8h1v1H14zM16 8h1v1H16zM18 8h1v1H18zM20,8 h1v1H20zM2 9h1v1H2zM8 9h3v1H8zM12 9h2v1H12zM16 9h1v1H16zM19,9 h2v1H19zM2 10h5v1H2zM9 10h2v1H9zM13 10h1v1H13zM15 10h1v1H15zM18 10h2v1H18zM3 11h1v1H3zM5 11h1v1H5zM8 11h1v1H8zM10 11h1v1H10zM12 11h4v1H12zM18 11h1v1H18zM20,11 h1v1H20zM0 12h2v1H0zM3 12h1v1H3zM5 12h3v1H5zM9 12h1v1H9zM11 12h1v1H11zM13 12h2v1H13zM16 12h2v1H16zM20,12 h1v1H20zM8 13h3v1H8zM13 13h5v1H13zM19 13h1v1H19zM0 14h7v1H0zM10 14h2v1H10zM14 14h2v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h1v1H8zM11 15h1v1H11zM14 15h1v1H14zM18 15h2v1H18zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h5v1H9zM16 16h2v1H16zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h3v1H9zM13 17h2v1H13zM16 17h1v1H16zM18 17h1v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM9 18h2v1H9zM12 18h1v1H12zM15 18h3v1H15zM19,18 h2v1H19zM0 19h1v1H0zM6 19h1v1H6zM9 19h1v1H9zM11 19h1v1H11zM15 19h1v1H15zM18 19h1v1H18zM0 20h7v1H0zM13 20h1v1H13zM15 20h1v1H15zM17 20h1v1H17zM19 20h1v1H19z"
        fill="#000000"
      />
    `)
  })
})
