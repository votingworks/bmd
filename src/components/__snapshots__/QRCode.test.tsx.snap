// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QR Code renders QR Code with size 1`] = `
.c0 > svg {
  display: block;
  width: 100%;
  height: auto;
}

<span
  class="c0"
>
  <svg
    height="128"
    shape-rendering="crispEdges"
    viewBox="0 0 21 21"
    width="128"
  >
    <path
      d="M0,0 h21v21H0z"
      fill="#FFFFFF"
    />
    <path
      d="M0 0h7v1H0zM8 0h3v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM10 1h1v1H10zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM11 2h1v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h4v1H9zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM12 4h1v1H12zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM9 5h1v1H9zM11 5h2v1H11zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h1v1H8zM12 7h1v1H12zM5 8h2v1H5zM9 8h2v1H9zM14 8h1v1H14zM16 8h1v1H16zM18 8h1v1H18zM20,8 h1v1H20zM2 9h1v1H2zM8 9h3v1H8zM12 9h2v1H12zM16 9h1v1H16zM19,9 h2v1H19zM2 10h5v1H2zM9 10h2v1H9zM13 10h1v1H13zM15 10h1v1H15zM18 10h2v1H18zM3 11h1v1H3zM5 11h1v1H5zM8 11h1v1H8zM10 11h1v1H10zM12 11h4v1H12zM18 11h1v1H18zM20,11 h1v1H20zM0 12h2v1H0zM3 12h1v1H3zM5 12h3v1H5zM9 12h1v1H9zM11 12h1v1H11zM13 12h2v1H13zM16 12h2v1H16zM20,12 h1v1H20zM8 13h3v1H8zM13 13h5v1H13zM19 13h1v1H19zM0 14h7v1H0zM10 14h2v1H10zM14 14h2v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h1v1H8zM11 15h1v1H11zM14 15h1v1H14zM18 15h2v1H18zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h5v1H9zM16 16h2v1H16zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h3v1H9zM13 17h2v1H13zM16 17h1v1H16zM18 17h1v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM9 18h2v1H9zM12 18h1v1H12zM15 18h3v1H15zM19,18 h2v1H19zM0 19h1v1H0zM6 19h1v1H6zM9 19h1v1H9zM11 19h1v1H11zM15 19h1v1H15zM18 19h1v1H18zM0 20h7v1H0zM13 20h1v1H13zM15 20h1v1H15zM17 20h1v1H17zM19 20h1v1H19z"
      fill="#000000"
    />
  </svg>
</span>
`;

exports[`renders QR Code renders basic QR Code 1`] = `
.c0 > svg {
  display: block;
  width: 1in;
  height: auto;
}

<span
  class="c0"
>
  <svg
    height="128"
    shape-rendering="crispEdges"
    viewBox="0 0 29 29"
    width="128"
  >
    <path
      d="M0,0 h29v29H0z"
      fill="#FFFFFF"
    />
    <path
      d="M0 0h7v1H0zM9 0h2v1H9zM13 0h1v1H13zM15 0h3v1H15zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM9 1h5v1H9zM15 1h3v1H15zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM9 2h2v1H9zM13 2h2v1H13zM18 2h2v1H18zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h2v1H9zM12 3h1v1H12zM15 3h3v1H15zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM11 4h1v1H11zM13 4h1v1H13zM15 4h1v1H15zM19 4h1v1H19zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM10 5h2v1H10zM14 5h1v1H14zM17 5h3v1H17zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM8 7h1v1H8zM15 7h1v1H15zM2 8h2v1H2zM6 8h3v1H6zM10 8h1v1H10zM12 8h1v1H12zM15 8h2v1H15zM18 8h2v1H18zM21 8h2v1H21zM24 8h1v1H24zM1 9h3v1H1zM7 9h1v1H7zM9 9h1v1H9zM11 9h2v1H11zM14 9h1v1H14zM18 9h1v1H18zM20 9h1v1H20zM22 9h5v1H22zM28,9 h1v1H28zM3 10h4v1H3zM8 10h2v1H8zM14 10h1v1H14zM16 10h1v1H16zM19 10h4v1H19zM24 10h1v1H24zM27 10h1v1H27zM2 11h2v1H2zM5 11h1v1H5zM14 11h1v1H14zM17 11h2v1H17zM20 11h2v1H20zM24 11h1v1H24zM28,11 h1v1H28zM4 12h5v1H4zM10 12h1v1H10zM12 12h10v1H12zM25,12 h4v1H25zM0 13h1v1H0zM2 13h2v1H2zM5 13h1v1H5zM10 13h1v1H10zM12 13h1v1H12zM15 13h3v1H15zM19 13h1v1H19zM22 13h2v1H22zM25 13h2v1H25zM28,13 h1v1H28zM0 14h1v1H0zM6 14h2v1H6zM10 14h3v1H10zM15 14h3v1H15zM19 14h2v1H19zM22 14h1v1H22zM24 14h2v1H24zM28,14 h1v1H28zM0 15h2v1H0zM4 15h1v1H4zM8 15h1v1H8zM11 15h1v1H11zM14 15h3v1H14zM18 15h3v1H18zM22 15h1v1H22zM25 15h1v1H25zM28,15 h1v1H28zM0 16h2v1H0zM6 16h7v1H6zM18 16h2v1H18zM23 16h3v1H23zM27 16h1v1H27zM3 17h1v1H3zM5 17h1v1H5zM7 17h2v1H7zM10 17h1v1H10zM13 17h1v1H13zM17 17h1v1H17zM19 17h1v1H19zM22 17h2v1H22zM0 18h1v1H0zM5 18h4v1H5zM12 18h1v1H12zM15 18h1v1H15zM17 18h2v1H17zM20 18h1v1H20zM22 18h1v1H22zM25 18h2v1H25zM3 19h3v1H3zM7 19h1v1H7zM11 19h2v1H11zM15 19h3v1H15zM20 19h4v1H20zM25,19 h4v1H25zM1 20h2v1H1zM4 20h3v1H4zM8 20h2v1H8zM13 20h1v1H13zM19 20h8v1H19zM28,20 h1v1H28zM8 21h1v1H8zM10 21h2v1H10zM13 21h1v1H13zM16 21h2v1H16zM20 21h1v1H20zM24 21h2v1H24zM27,21 h2v1H27zM0 22h7v1H0zM8 22h2v1H8zM11 22h2v1H11zM16 22h5v1H16zM22 22h1v1H22zM24 22h1v1H24zM26 22h2v1H26zM0 23h1v1H0zM6 23h1v1H6zM9 23h1v1H9zM11 23h2v1H11zM14 23h1v1H14zM16 23h2v1H16zM20 23h1v1H20zM24 23h1v1H24zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM10 24h1v1H10zM12 24h1v1H12zM14 24h3v1H14zM20 24h7v1H20zM28,24 h1v1H28zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM8 25h1v1H8zM10 25h1v1H10zM13 25h1v1H13zM15 25h1v1H15zM17 25h2v1H17zM21 25h1v1H21zM24 25h2v1H24zM27 25h1v1H27zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM8 26h2v1H8zM14 26h1v1H14zM16 26h2v1H16zM21 26h1v1H21zM26 26h1v1H26zM28,26 h1v1H28zM0 27h1v1H0zM6 27h1v1H6zM10 27h2v1H10zM13 27h1v1H13zM22 27h1v1H22zM24 27h2v1H24zM27 27h1v1H27zM0 28h7v1H0zM9 28h3v1H9zM14 28h2v1H14zM17 28h1v1H17zM19 28h1v1H19zM22 28h2v1H22zM25 28h1v1H25zM27 28h1v1H27z"
      fill="#000000"
    />
  </svg>
</span>
`;

exports[`renders QRCode 1`] = `
.c0 > svg {
  display: block;
  width: 1in;
  height: auto;
}

<span
  class="c0"
>
  <svg
    height="128"
    shape-rendering="crispEdges"
    viewBox="0 0 21 21"
    width="128"
  >
    <path
      d="M0,0 h21v21H0z"
      fill="#FFFFFF"
    />
    <path
      d="M0 0h7v1H0zM8 0h3v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM10 1h1v1H10zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM11 2h1v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h4v1H9zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM12 4h1v1H12zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM9 5h1v1H9zM11 5h2v1H11zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h1v1H8zM12 7h1v1H12zM5 8h2v1H5zM9 8h2v1H9zM14 8h1v1H14zM16 8h1v1H16zM18 8h1v1H18zM20,8 h1v1H20zM2 9h1v1H2zM8 9h3v1H8zM12 9h2v1H12zM16 9h1v1H16zM19,9 h2v1H19zM2 10h5v1H2zM9 10h2v1H9zM13 10h1v1H13zM15 10h1v1H15zM18 10h2v1H18zM3 11h1v1H3zM5 11h1v1H5zM8 11h1v1H8zM10 11h1v1H10zM12 11h4v1H12zM18 11h1v1H18zM20,11 h1v1H20zM0 12h2v1H0zM3 12h1v1H3zM5 12h3v1H5zM9 12h1v1H9zM11 12h1v1H11zM13 12h2v1H13zM16 12h2v1H16zM20,12 h1v1H20zM8 13h3v1H8zM13 13h5v1H13zM19 13h1v1H19zM0 14h7v1H0zM10 14h2v1H10zM14 14h2v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h1v1H8zM11 15h1v1H11zM14 15h1v1H14zM18 15h2v1H18zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h5v1H9zM16 16h2v1H16zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h3v1H9zM13 17h2v1H13zM16 17h1v1H16zM18 17h1v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM9 18h2v1H9zM12 18h1v1H12zM15 18h3v1H15zM19,18 h2v1H19zM0 19h1v1H0zM6 19h1v1H6zM9 19h1v1H9zM11 19h1v1H11zM15 19h1v1H15zM18 19h1v1H18zM0 20h7v1H0zM13 20h1v1H13zM15 20h1v1H15zM17 20h1v1H17zM19 20h1v1H19z"
      fill="#000000"
    />
  </svg>
</span>
`;
